src/AhRulesBot/AhRulesBot/BotRequestsProcessing/MessageValidator.cs

```diff
@@ -46,21 +54,49 @@ public async Task<bool> IsValid(Message message)
             if (DateTime.UtcNow.Subtract(message.Date).TotalMinutes > 4)
                 return false;
 
+            if (IsSpamming(message))
+            {
+                return false;
+            }
+
             return true;
         }
 
+        private bool IsSpamming(Message message)
+        {
+            var bannedCacheKey = $"{message.From.Id}";
+            var requestCacheKey = $"{message.From.Id}_{DateTime.UtcNow.ToString("H: mm", CultureInfo.InvariantCulture)}";
+
+            if (_bannedCache.TryGetValue(bannedCacheKey, out bool isBanned) && isBanned)
+            {
+                return true;
+            }
+
+            _requestsCache.TryGetValue(requestCacheKey, out int value);
+            value++;
+            _requestsCache.Set(requestCacheKey, value, new MemoryCacheEntryOptions().SetAbsoluteExpiration(_requestCacheTimeout));
```

Comment:
I would not call that `*Timeout`, suggest to call it `expiration`
Timeouts are for operations, like if you would use redis cache here and call it over the network - that request would have some of timeout. And the timeout would cancel the request if redis is down/unavailable. 
Expiration is different, it shows the lifetime of your data in a cache